src/data/abilities/ab-attrs/full-hp-resist-type-ab-attr.ts

```diff
@@ -24,27 +24,22 @@ export class FullHpResistTypeAbAttr extends PreDefendAbAttr {
    * @param attacker n/a
    * @param move {@linkcode Move} the move being used on the source
    * @param typeMultiplier a container for the move's current type effectiveness multiplier
-   * @returns `true` if the move's effectiveness is reduced; `false` otherwise
    */
   public override apply(
     pokemon: Pokemon,
     simulated: boolean,
     _attacker: Pokemon,
-    move: Move,
-    typeMultiplier: NumberHolder,
-  ): boolean {
-    if (move?.hasAttr(FixedDamageAttr)) {
-      return false;
+    _move: Move,
+    typeMultiplier: ValueHolder<number>,
+  ): void {
+    typeMultiplier.value = 0.5;
+    if (!simulated) {
+      pokemon.turnData.moveEffectiveness = 0.5;
     }
+  }
 
-    if (pokemon.isFullHp() && typeMultiplier.value > 0.5) {
-      typeMultiplier.value = 0.5;
-      if (!simulated) {
-        pokemon.turnData.moveEffectiveness = 0.5;
-      }
-      return true;
-    }
-    return false;
+  public override canApply(...[pokemon, , , move, typeMultiplier]: Parameters<this["apply"]>): boolean {
+    return pokemon.isFullHp() && !move.hasAttr(FixedDamageAttr) && typeMultiplier.value > 0.5;
```

Comment:
Just a small micro-optimization
```suggestion
    return pokemon.isFullHp() && typeMultiplier.value > 0.5 && !move.hasAttr(FixedDamageAttr);
```